test/e2e/node_feature_discovery_test.go

```diff
@@ -873,6 +887,62 @@ core:
 				})
 			})
 
+			// Test NodeFeatureGroups
+			Context("and NodeFeatureGroups objects deployed", Label("nodefeaturegroup"), func() {
+				BeforeEach(func(ctx context.Context) {
+					// We need a NodeFeature from the node, can't be a fake one
+					if !useNodeFeatureApi {
+						Skip("NodeFeature API not enabled")
+					}
+					// enable the node feature group api
+					extraMasterPodSpecOpts = []testpod.SpecOption{
+						testpod.SpecWithContainerExtraArgs(
+							"--feature-gates=NodeFeatureGroupAPI=true",
+						),
+					}
+				})
+				It("custom NodeFeatureGroup should be updated", func(ctx context.Context) {
+					By("Creating nfd-worker daemonset")
+					podSpecOpts := createPodSpecOpts(
+						testpod.SpecWithContainerImage(dockerImage()),
+					)
+					workerDS := testds.NFDWorker(podSpecOpts...)
+					workerDS, err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Create(ctx, workerDS, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Waiting for worker daemonset pods to be ready")
+					Expect(testpod.WaitForReady(ctx, f.ClientSet, f.Namespace.Name, workerDS.Spec.Template.Labels["name"], 2)).NotTo(HaveOccurred())
+
+					nodes, err := getNonControlPlaneNodes(ctx, f.ClientSet)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Creating NodeFeatureGroups #1")
+					Expect(testutils.CreateNodeFeatureGroupsFromFile(ctx, nfdClient, f.Namespace.Name, "nodefeaturegroup-1.yaml")).NotTo(HaveOccurred())
+
+					By("Verifying NodeFeatureGroups #1")
+					targetNodes := make([]string, 0)
+					for _, node := range nodes {
+						if !Expect(node.Name).ToNot(BeEmpty(), "No suitable worker node found") {
+							continue
+						}
```

Comment:
I still don't get this. In which case would `node.Name` be empty? Why just not `targetNodes = nodes` -> drop the targetNodes var and just use nodes?

Reply:
NFG.Status.Nodes is a list of strings, so we only need the name, but you are right, I can trim it a bit